### ISSUE TITLE
A way to select commits based on PRs labels?

### ISSUE THREAD

**Issue by ariselseng** [sqren/backport#184]
Hello,
Thank you for this awesome utility. It will definitly save me a lot of time.

I want to pre-select multiple commits based on the labels of the original PR. Is that possible to day?

There seems to be a  --backportCreatedLabels but it does not seem to do anything.

**Comment by sqren** [comment]
Hi,

Thanks for reaching out! 
Can you please elaborate on your usecase a little bit so I can get a better understanding. 
If you run `backport --pr 123` I'd expect you want to backport the commits for that PR - and not some other commits based on a label?

**Comment by ariselseng** [comment]
My use case would be to backport multiple pull requests that has the label "backport-v2". So if there there was a flag that would filter the list of commits to only list commits of pull requests that have the specified label it would solve my use case I think. 
Using the flag "pr" and this flag would not work at the same time.

**Comment by sqren** [comment]
Ahh, I see. That makes sense. 

Would this be something you'd mostly want to do ad-hoc (via cli)?
```
backport --filter.labels backport-v2
```

Or a semi-permanent thing you'd set via config?
```json
{
  "sourcePRFilter": {
    "labels": ["backport-v2"]
  }
}
```

To stay consistent with the current behaviour I'd prefer simply listing the commits that matches the filter (and not pre-selecting them). Would that work for you?

**Comment by ariselseng** [comment]
Yes that makes sense :)

**Comment by ariselseng** [comment]
I think a cli flag would best fit my use case at least

**Comment by sqren** [comment]
I've released this feature as a beta version: 5.4.0-beta.1. Try it out when you have time, and then let me know how it goes.

Install the beta:
```
npm install backport@5.4.0-beta.1 
```

List PRs with label "backport-v2":
```
backport --pr-filter label:backport-v2
```

`--pr-filter` accepts any valid Github search qualifier. So you can also find any PR that mentions a specific person:

```
backport --pr-filter mentions:ariselseng
```

You can also mix qualifiers:
```
backport --pr-filter "reactions:>10 label:backport-v2"
```

By default you will only see your own PRs. To see PRs from anyone use `--all`:
```
backport --pr-filter label:backport-v2 --all
```

Check the docs for more examples:
https://help.github.com/en/github/searching-for-information-on-github/searching-issues-and-pull-requests

**Comment by ariselseng** [comment]
@sqren I just tested the beta. It works great!

**Comment by ariselseng** [comment]
@sqren I just tried the beta with the full flow. And it created a PR in my repo, but it hanged on "Pushing branch backport/release/pr-1867".
It even hanged when using standard "--pr 1867" parameter.
Reverting back to release 5.2.0 solves the issue.

**Comment by sqren** [comment]
Hi @ariselseng 

Good feedback. I already fixed that bug but forgot to publish. Thanks for reminding me :)

It's available in 5.4.0-beta.2

**Comment by sqren** [comment]
Released in 5.4.0
